### PR TITLE
Issue 336 transition to semver

### LINKED ISSUE #336: Transition versioning to semver
Including/mainly versioncheck URL variables

    15:24 < devgeeks> 0.0.2 -> 0.0.3 should be added features, but backwards compatible or something.. but 0.0.2 -> 0.1.0 should be a client complainer

### PR DESCRIPTION
**Pull request by devgeeks**
Consolidates various versions and adds semver

Fixes #336

There were three different versions hardcoded accross various files.

This change makes `package.json` the "source of truth" as well as having
the `versioncheck` use semver to only return false on breaking changes.

### CODE REVIEW

**Comment by daviddahl** [comment]
quick link to failed test: https://travis-ci.org/SpiderOak/crypton/builds/46157962#L2228

**Comment by devgeeks** [comment]
Gah. I didn't see any server side tests, but didn't even think about if there would be a client side one already.

I'll fix it up.

File: client/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "author": "SpiderOak (https://spideroak.com/)",
   "name": "crypton",
   "description": "A zero-knowledge application framework",
-  "version": "0.0.2",
+  "version": "0.0.4",
```
Comment (daviddahl) [inline client/package.json]: Version 4? Wut?

Comment (devgeeks) [inline client/package.json]: The server's version in `package.json` was already `0.0.4`, shooting for consistency.

**Pull request merged**